docs/vm-metrics.md

```diff
@@ -0,0 +1,178 @@
+| virtual CPU #x state               | hypervisor    | -         | -           |
+| virtual CPU #x total time          | hypervisor    | -         | -           |
+| network interface count            | hypervisor    | -         | -           |
+| block device count                 | hypervisor    | -         | -           |
+| migration downtime                 | hypervisor    | yes       |             |
+| migration elapsed time             | hypervisor    | yes       |             |
+| migration total data               | hypervisor    | yes       |             |
+| migration processed data           | hypervisor    | yes       |             |
+| migration remaining data           | hypervisor    | yes       |             |
+| migration total memory             | hypervisor    | yes       |             |
+| migration processed memory         | hypervisor    | yes       |             |
+| migration remaining memory         | hypervisor    | yes       |             |
+| migration memory bps               | hypervisor    | yes       |             |
+| migration memory constant [TODO]   | hypervisor    | yes       |             |
+| migration memory compression bytes | hypervisor    | yes       |             |
+| migration dirty rate               | hypervisor    | yes       |             |
+| migration memory iteration         | hypervisor    | yes       |             |
+| guest total memory                 | guest agent   | -         |             |
+| guest free memory                  | guest agent   | -         |             |
+| guest CPU count                    | guest agent   | -         |             |
+| guest free disk space              | guest agent   | -         |             |
+| guest IP address                   | guest agent   | -         |             |
+
+
+**NOTE** the list of guest agent provided metrics is partial, because
+a lot of different metrics can be collected for the guest. The presented
+entries are a subset of what the [QEMU Guest Agent](https://wiki.qemu.org/Features/GuestAgent)
+provides, which is used by virtualization managers like
+[oVirt](http://www.ovirt.org)
+
+From now on, the overlapping metrics  (aka 'system metrics') will be ignored,
+and we will focus on the 'hypervisor' and 'guest agent' metrics.
+
+## collection of VM metrics
+A natural fit for the kubevirt specific metrics is the [custom metrics](
+https://github.com/kubernetes/kubernetes/blob/release-1.2/docs/proposals/custom-metrics.md).
+All the metrics are accessible through the libvirt API.
+
+The collection of the VM metrics will be done by a separate pod, implemented as kubernetes DaemonSet.
+This is because the metrics are available from a centralized source, the libvirt daemon.
+To increase resilience, each node should expose all the metrics it knows about, avoiding
+a single point of failure per cluster. The resulting architecture is shown in the figure below.
+
+![VM metrics diagram](vm-metrics.png "VM Metrics")
```

Comment:
wrong diagram, I (hastly) copied another one and got it wrong. Libvirt and vAdvisor must run on each node. Will update it ASAP.